xbmc/games/dialogs/osd/DialogGameVideoFilter.cpp

```diff
@@ -250,19 +285,101 @@ void CDialogGameVideoFilter::PostExit()
 
 bool CDialogGameVideoFilter::OnClickAction()
 {
+  if (static_cast<int>(m_focusedItemIndex) < m_items.Size())
+  {
+    const std::shared_ptr<CFileItem> focusedItem = m_items[m_focusedItemIndex];
+
+    // "Get more..." button has an empty string for the video filter
+    if (focusedItem && focusedItem->GetProperty("game.videofilter").empty())
+    {
+      OnGetMore();
+      return true;
+    }
+  }
+
   Close();
   return true;
 }
 
+void CDialogGameVideoFilter::RefreshList()
+{
+  // Re-generate the items in case the shader preset add-on was installed
+  if (m_regenerateList)
+  {
+    m_regenerateList = false;
+    PreInit();
+  }
+
+  // Call ancestor
+  CDialogGameVideoSelect::RefreshList();
+}
+
 std::string CDialogGameVideoFilter::GetLocalizedString(uint32_t code)
 {
   return g_localizeStrings.GetAddonString(PRESETS_ADDON_NAME, code);
 }
 
-void CDialogGameVideoFilter::GetProperties(const CFileItem& item,
-                                           std::string& videoFilter,
-                                           std::string& description)
+void CDialogGameVideoFilter::GetProperties(const CFileItem& item, std::string& videoFilter)
 {
   videoFilter = item.GetProperty("game.videofilter").asString();
-  description = item.GetProperty("game.videofilterdescription").asString();
+}
+
+void CDialogGameVideoFilter::OnGetMore()
+{
+  std::shared_ptr<CJobManager> jobManager = CServiceBroker::GetJobManager();
+  if (!jobManager)
+    return;
+
+  jobManager->Submit(
+      [this]()
+      {
+        using namespace ADDON;
+
+        CAddonMgr& addonManager = CServiceBroker::GetAddonMgr();
+
+        bool success = false;
+        if (addonManager.IsAddonDisabled(PRESETS_ADDON_NAME))
+        {
+          success = addonManager.EnableAddon(PRESETS_ADDON_NAME);
+        }
+        else if (!addonManager.IsAddonInstalled(PRESETS_ADDON_NAME))
+        {
+          AddonPtr addon;
+          success = CAddonInstaller::GetInstance().InstallModal(PRESETS_ADDON_NAME, addon,
+                                                                InstallModalPrompt::CHOICE_NO);
+        }
+
+        if (success)
+          OnGetMoreComplete();
+      });
+}
+
+void CDialogGameVideoFilter::OnGetMoreComplete()
+{
+  CGUIComponent* gui = CServiceBroker::GetGUI();
+  if (gui == nullptr)
+    return;
+
+  // Shader preset add-ons are loaded async, so wait until the new
+  // add-on is fully loaded. Bail after a reasonable timeout to avoid
+  // waiting indefinitely.
+  XbmcThreads::EndTime<> timeout(5s);
+  while (!CServiceBroker::GetGameServices().VideoShaders().HasAddons() && !timeout.IsTimePast())
```

Comment:
Reverse the terms to save some CPU cycles? `HasAddons` is  quite expensive, IIRC and no need to call if timed out.